course3/homework-string/my_string.h

```diff
@@ -1,6 +1,110 @@
 #pragma once
+#include <iostream>
+#include <memory>
 
 class MyString
 {
-	// TODO
+private:
+    std::size_t length;
+    std::unique_ptr<char []> buff;
+
+public:
+    MyString() : length(0) {}
+    
+    MyString(const char *data) : length(0), buff() {
+        length = std::strlen(data);
+
+        buff = std::make_unique<char []>(length + 1);        
+
+        std::strcpy(buff.get(), data);
+    }
+
+    MyString(const MyString& rhs) : length(0), buff() {
+        length = rhs.size();
+        if (rhs.buff.get()) {
+            buff = std::make_unique<char []>(length + 1);        
+            std::strcpy(buff.get(), rhs.buff.get());
+        }
+    }
+
+    MyString(MyString&& rhs) : length(rhs.size()), buff(std::move(rhs.buff)) { }
+
+    bool empty() const {
+        return length == 0;
+    }
+
+    std::size_t size() const {
+        return length;
+    }
+
+    const char* c_str() const {
+        if (buff)
+            return buff.get();
+        else
+            return "";
+    }
+
+    MyString& operator=(const MyString& rhs) {
+        if (rhs.empty()) {
+            buff.reset();
+        }
+        else {
+            buff = std::make_unique<char []>(rhs.length + 1);
+            std::strcpy(buff.get(), rhs.buff.get());
```

Comment:
What about length? Punitive excercise: write a unit test that will fail because of this bug, and only then fix it.